android-huawei/src/de/golfgl/gdxgamesvcs/HuaweiGameServicesClient.java

```diff
@@ -0,0 +1,671 @@
+
+import android.content.Intent;
+import android.text.TextUtils;
+
+import com.badlogic.gdx.backends.android.AndroidApplication;
+import com.badlogic.gdx.backends.android.AndroidEventListener;
+import com.badlogic.gdx.utils.Array;
+import com.huawei.hmf.tasks.OnFailureListener;
+import com.huawei.hmf.tasks.OnSuccessListener;
+import com.huawei.hmf.tasks.Task;
+import com.huawei.hms.common.ApiException;
+import com.huawei.hms.jos.JosApps;
+import com.huawei.hms.jos.JosAppsClient;
+import com.huawei.hms.jos.games.AchievementsClient;
+import com.huawei.hms.jos.games.ArchivesClient;
+import com.huawei.hms.jos.games.EventsClient;
+import com.huawei.hms.jos.games.GameScopes;
+import com.huawei.hms.jos.games.Games;
+import com.huawei.hms.jos.games.PlayersClient;
+import com.huawei.hms.jos.games.RankingsClient;
+import com.huawei.hms.jos.games.achievement.Achievement;
+import com.huawei.hms.jos.games.archive.Archive;
+import com.huawei.hms.jos.games.archive.ArchiveDetails;
+import com.huawei.hms.jos.games.archive.ArchiveSummary;
+import com.huawei.hms.jos.games.archive.ArchiveSummaryUpdate;
+import com.huawei.hms.jos.games.archive.OperationResult;
+import com.huawei.hms.jos.games.player.Player;
+import com.huawei.hms.support.api.entity.auth.Scope;
+import com.huawei.hms.support.hwid.HuaweiIdAuthManager;
+import com.huawei.hms.support.hwid.request.HuaweiIdAuthParams;
+import com.huawei.hms.support.hwid.request.HuaweiIdAuthParamsHelper;
+import com.huawei.hms.support.hwid.result.AuthHuaweiId;
+import com.huawei.hms.support.hwid.result.HuaweiIdAuthResult;
+import com.huawei.hms.support.hwid.service.HuaweiIdAuthService;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import de.golfgl.gdxgamesvcs.achievement.IAchievement;
+import de.golfgl.gdxgamesvcs.achievement.IFetchAchievementsResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.IFetchGameStatesListResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.ILoadGameStateResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.ISaveGameStateResponseListener;
+import de.golfgl.gdxgamesvcs.leaderboard.IFetchLeaderBoardEntriesResponseListener;
+import de.golfgl.gdxgamesvcs.leaderboard.ILeaderBoardEntry;
+
+/**
+ * Client for Huawei Game Services
+ * <p>
+ * Created by Francesco Stranieri on 24.07.2020.
+ */
+public class HuaweiGameServicesClient implements IGameServiceClient, AndroidEventListener {
+
+    private final int HUAWEI_GAMESVCS_AUTH_REQUEST = 8971;
+    private final int HUAWEI_GAMESVCS_ACHIEVEMENTS_REQUEST = 8972;
+    private final int HUAWEI_GAMESVCS_LEADERBOARDS_REQUEST = 8973;
+
+    private AndroidApplication activity;
+    private IGameServiceIdMapper<String> huaweiLeaderboardIdMapper;
+    private IGameServiceIdMapper<String> huaweiAchievementIdMapper;
+    private IGameServiceIdMapper<String> huaweiGameEventIdMapper;
+    private JosAppsClient josAppsClient;
+    private AchievementsClient achievementsClient;
```

Comment:
Suggestion: make all the clients `protected`. This way, subclasses cann access the clients in case some game need to adjust the behaviour.